ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/admin/ApiPackageManagementService.scala

```diff
@@ -40,76 +40,76 @@ final class ApiPackageManagementService private (
+      else
+        SubmissionId.assertFromString(request.submissionId)
 
-      // Amount of time we wait for the ledger to commit the request before we
-      // give up on polling for the result.
-      // TODO(JM): This constant should be replaced by user-provided maximum record time
-      // which should be wired through the stack and verified during validation, just like
-      // with transactions. I'm leaving this for another PR.
-      val timeToLive = 30.seconds
+    // Amount of time we wait for the ledger to commit the request before we
+    // give up on polling for the result.
+    // TODO(JM): This constant should be replaced by user-provided maximum record time
+    // which should be wired through the stack and verified during validation, just like
+    // with transactions. I'm leaving this for another PR.
+    val timeToLive = 30.seconds
 
-      implicit val ec: ExecutionContext = DE
-      for {
-        dar <- DarReader { case (_, x) => Try(Archive.parseFrom(x)) }
-          .readArchive(
-            "package-upload",
-            new ZipInputStream(new ByteArrayInputStream(request.darFile.toByteArray)))
-          .fold(
-            err => Future.failed(ErrorFactories.invalidArgument(err.getMessage)),
-            Future.successful
-          )
-        ledgerEndBeforeRequest <- transactionsService.currentLedgerEnd()
-        result <- FutureConverters.toScala(
-          packagesWrite.uploadPackages(submissionId, dar.all, None)
+    implicit val ec: ExecutionContext = DE
+    val response = for {
+      dar <- DarReader { case (_, x) => Try(Archive.parseFrom(x)) }
+        .readArchive(
+          "package-upload",
+          new ZipInputStream(new ByteArrayInputStream(request.darFile.toByteArray)))
+        .fold(
+          err => Future.failed(ErrorFactories.invalidArgument(err.getMessage)),
+          Future.successful
         )
-        response <- result match {
-          case SubmissionResult.Acknowledged =>
-            pollUntilPersisted(submissionId, timeToLive, ledgerEndBeforeRequest).flatMap {
-              case _: PackageEntry.PackageUploadAccepted =>
-                Future.successful(UploadDarFileResponse())
-              case PackageEntry.PackageUploadRejected(_, _, reason) =>
-                Future.failed(ErrorFactories.invalidArgument(reason))
-            }
-          case r @ SubmissionResult.Overloaded =>
-            Future.failed(ErrorFactories.resourceExhausted(r.description))
-          case r @ SubmissionResult.InternalError(_) =>
-            Future.failed(ErrorFactories.internal(r.reason))
-          case r @ SubmissionResult.NotSupported =>
-            Future.failed(ErrorFactories.unimplemented(r.description))
-        }
-      } yield response
-    }
+      ledgerEndBeforeRequest <- transactionsService.currentLedgerEnd()
+      result <- FutureConverters.toScala(
+        packagesWrite.uploadPackages(submissionId, dar.all, None)
+      )
+      result <- result match {
```

Comment:
Perhaps avoid shadowing here by renaming this `result` to `completed` or something?

Reply:
Didn't notice that, thanks.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4097/commits/80ef9a78495c345ca3a7442bc7c2397f4901c132